Geometry_Engine/Query/GeometryHash.cs

```diff
@@ -0,0 +1,427 @@
+            "\nThe hash is computed as an array representing the coordinate of significant points taken on the geometry." +
+            "\nThe number of points is reduced to the minimum essential to determine uniquely any geometry." +
+            "\nAdditionally, the resulting points are transformed based on the source geometry type, to remove or minimize collisions." +
+            "\n(Any transformation so performed is translational only, in order to support geometrical tolerance, i.e. numerical distance, when comparing GeometryHashes downstream).")]
+        [Output("geomHash", "Array of numbers representing a unique signature of the input geometry.")]
+        public static double[] IGeometryHash(this IGeometry igeometry)
+        {
+            return GeometryHash(igeometry as dynamic, 0);
+        }
+
+        /***************************************************/
+        /****              Private Methods              ****/
+        /***************************************************/
+
+
+        /***************************************************/
+        /****  Curves                                   ****/
+        /***************************************************/
+
+        [Description("The geometry hash of a Curve is obtained by first retrieving any Sub-part of the curve, if present." +
+            "The ISubParts() methods is able to return the 'primitive' curves that a curve is composed of. " +
+            "The GeometryHashes are then calculated for the individual parts and concatenated.")]
+        private static double[] GeometryHash(this ICurve curve, double translationFactor)
+        {
+            IEnumerable<ICurve> subParts = curve.ISubParts();
+
+            bool isMultipleCurves = subParts.Count() != 1;
+
+            return subParts.SelectMany(c => (double[])GeometryHash(c as dynamic, translationFactor, isMultipleCurves)).ToArray();
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for an Arc is calculated as the GeometryHash of the start, end and middle point of the Arc.")]
+        private static double[] GeometryHash(this Arc curve, double translationFactor, bool skipLastPoint = false)
+        {
+            translationFactor += (int)TypeTranslationFactor.Arc;
+
+            IEnumerable<double> hash = curve.StartPoint().ToDoubleArray(translationFactor)
+               .Concat(curve.PointAtParameter(0.5).ToDoubleArray(translationFactor));
+
+            if (!skipLastPoint)
+                hash = hash.Concat(curve.EndPoint().ToDoubleArray(translationFactor));
+
+            return hash.ToArray();
+        }
+
+        /***************************************************/
+
+        [Description("The GeometryHash for an Circle is calculated as the GeometryHash of the start, 1/3rd and 2/3rd points of the Circle.")]
+        private static double[] GeometryHash(this Circle curve, double translationFactor, bool skipLastPoint = false)
```

Comment:
```suggestion
        private static double[] GeometryHash(this Circle curve, double translationFactor, bool skipEndPoint = false)
```

Would probably, as written on the arc above, change this to skipEndPoint and have it do nothing for the case of circle and Ellipse.

Not sure it makes sense to skip a point for these kinds of curve, and by renaming it with the EndPoint instead of Last, it can be semantically more reasonable to just have the argument do nothing, as a circle and ellipse kind of by definition do not have endpoints.